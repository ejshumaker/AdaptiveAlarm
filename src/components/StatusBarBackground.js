import React from 'react';
import {
  View, StyleSheet, Platform, NativeModules,
} from 'react-native';

const { StatusBarManager } = NativeModules;

const styles = StyleSheet.create({
  statusBarBackground: {
    height: (Platform.OS === 'ios') ? 18 : StatusBarManager.HEIGHT, // this is just to test if the platform is iOS to give it a height of 18, else, no height (Android apps have their own status bar)
    backgroundColor: 'rgba(255,255,255,0)',
  },
});

const StatusBarBackground = (props) => {
  // eslint-disable-next-line
  const { style } = props;
  return (
    <View style={[styles.statusBarBackground, style || {}]} />
  );
};


export default StatusBarBackground;
