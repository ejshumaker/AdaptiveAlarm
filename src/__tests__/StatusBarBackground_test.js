/* eslint-disable no-undef */
/* eslint-disable import/first */
import 'react-native';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import StatusBarBackground from '../components/StatusBarBackground';

// jest.mock('react-native-svg');

Enzyme.configure({ adapter: new Adapter() });

jest.setTimeout(10000);
let wrapper = null;
const onPressMock = jest.fn();


describe('Status Bar tests', () => {
  beforeEach(() => {
    wrapper = mount(<StatusBarBackground
      onPress={onPressMock}
      title="test"
      textColor="b"
      backgroundColor="w"
    />);
    jest.clearAllMocks();
  });

  it('test StatusBarBackground matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
