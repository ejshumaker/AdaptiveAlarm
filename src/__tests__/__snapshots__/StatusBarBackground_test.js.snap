// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Bar tests test StatusBarBackground matches snapshot 1`] = `
<StatusBarBackground
  backgroundColor="w"
  onPress={[MockFunction]}
  textColor="b"
  title="test"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255,255,255,0)",
          "height": 18,
        },
        Object {},
      ]
    }
  >
    <react-native-mock />
  </View>
</StatusBarBackground>
`;
